src/components/text-editor/examples/text-editor-custom-triggers.tsx

```diff
@@ -71,59 +86,90 @@ export class TextEditorCustomTriggersExample {
+    private setupEventHandlers() {
         if (this.isPickerOpen) {
-            document.addEventListener('click', this.globalClickListener, {
+            this.host.addEventListener('click', this.handleClick, {
+                capture: true,
+            });
+            this.host.addEventListener('keydown', this.handleKeyPress, {
                 capture: true,
             });
-            document.addEventListener('keyup', this.handleGlobalKeyPress);
         } else {
-            document.removeEventListener('click', this.globalClickListener);
-            document.removeEventListener('keyup', this.handleGlobalKeyPress);
+            this.host.removeEventListener('click', this.handleClick, {
+                capture: true,
+            });
+            this.host.removeEventListener('keydown', this.handleKeyPress, {
+                capture: true,
+            });
         }
     }
 
-    private handleGlobalKeyPress = (event: KeyboardEvent) => {
-        if (event.key !== ESCAPE) {
-            return;
+    private handleKeyPress = (event: KeyboardEvent) => {
+        const capturedKeys = [ESCAPE, ARROW_UP, ARROW_DOWN, ENTER, TAB];
+        if (capturedKeys.includes(event.key)) {
+            event.stopPropagation();
+            event.preventDefault();
         }
 
-        event.stopPropagation();
-        event.preventDefault();
-        this.close.emit();
+        if (event.key === ARROW_DOWN || event.key === ARROW_UP) {
+            this.moveSelection(event.key);
+        }
+
+        if (event.key === ENTER || event.key === TAB) {
+            const selectedItem: ListItem | undefined = this.items.find(
+                (item) => item.selected,
+            );
+
+            if (selectedItem) {
+                this.insertItem(selectedItem);
+            }
+        }
+
+        if (event.key === ESCAPE) {
+            this.close.emit();
+        }
     };
 
-    private insertModeButtons: Button[] = [
-        {
-            id: '1',
-            title: 'text',
-            selected: true,
-        },
-        {
-            id: '2',
-            title: 'chip',
-        },
-    ];
+    private moveSelection = (
+        direction: typeof ARROW_UP | typeof ARROW_DOWN,
+    ) => {
+        const increment = direction === ARROW_DOWN ? 1 : -1;
+        const numberOfItems = this.items.length;
+        this.selectionIndex =
+            (this.selectionIndex + increment + numberOfItems) % numberOfItems;
 
-    private portalId: string;
-    private items: Array<ListItem<number>> = [
-        { text: 'Wolverine', value: 1, icon: 'wolf' },
-        { text: 'Captain America', value: 2, icon: 'captain_america' },
-        { text: 'Superman', value: 3, icon: 'superman' },
-        { text: 'Tony Stark', value: 4, icon: 'iron_man' },
-        { text: 'Batman', value: 5, icon: 'batman_old' },
-    ];
+        this.items = this.items.map((currentItem, index) => {
+            currentItem.selected = index === this.selectionIndex;
```

Comment:
The selection needs to based on the filtered items, not on allItems.

https://github.com/user-attachments/assets/cd2b6d6e-8007-47f6-bfba-2218ef085b55

Reply:
Thnx, see https://github.com/Lundalogik/lime-elements/pull/3316/commits/57f424181ad9b720c73e6b2c96081acecc5a0e14